react/features/base/dialog/components/native/ConfirmDialog.js

```diff
@@ -1,107 +1,171 @@
-     * translated using the provided params. See i18n function
-     * {@code translate(string, Object)} for more details.
+     * Dialog description key for translations.
+     */
+    descriptionKey?: string,
+
+    /**
+     * The i18n key of the text label for the cancel button.
+     */
+    cancelLabel: string,
+
+    /**
+     * The i18n key of the text label for the confirm button.
      */
-    contentKey: string | { key: string, params: Object},
+    confirmLabel: string,
 
     /**
-     * The handler for the event when clicking the 'confirmNo' button.
-     * Defaults to onCancel if absent.
+     * Callback to invoke when cancel button is clicked/pressed.
      */
-    onDecline?: Function,
+    onCancel: Function,
 
+    /**
+     * Callback to invoke when submit button is clicked/pressed.
+     */
+    onSubmit: Function,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
     t: Function
-}
+};
 
 /**
- * Implements a confirm dialog component.
+ * React Component for getting confirmation to stop a file recording session in
+ * progress.
+ *
+ * @augments Component
  */
-class ConfirmDialog extends BaseSubmitDialog<Props, *> {
+class ConfirmDialog<P: Props> extends BaseDialog<Props> {
     /**
-     * Returns the title key of the submit button.
+     * Initializes a new {@code ConfirmDialog} instance.
      *
-     * @returns {string}
+     * @inheritdoc
      */
-    _getSubmitButtonKey() {
-        return this.props.okKey || 'dialog.confirmYes';
+    constructor(props: P) {
+        super(props);
+
+        this.state = {
+            visible: true
+        };
+
+        // Bind event handler so it is only bound once for every instance.
+        this._onCancel = this._onCancel.bind(this);
+        this._onConfirm = this._onConfirm.bind(this);
     }
 
     _onCancel: () => void;
 
     /**
-     * Renders the 'No' button.
+     * Callback to cancel button.
      *
-     * NOTE: The {@code ConfirmDialog} is the only dialog right now that
-     * renders 2 buttons, mainly for clarity.
+     * @private
+     * @returns {void}
+     */
+    _onCancel() {
+        const { onCancel } = this.props;
+
+        onCancel && onCancel();
+
+        this.setState({
+            visible: !this.state.visible
+        });
+    }
+
+    _onConfirm: () => void;
+
+    /**
+     * Callback for the confirm button.
      *
-     * @inheritdoc
+     * @private
+     * @returns {void}.
+     */
+    _onConfirm() {
+        const { onSubmit } = this.props;
+
+        onSubmit && onSubmit();
+
+        this.setState({
+            visible: !this.state.visible
+        });
+    }
+
+    /**
+     * Renders the dialog description.
+     *
+     * @returns {React$Component}
      */
-    _renderAdditionalButtons() {
-        const { _dialogStyles, cancelKey, onDecline, t } = this.props;
+    _renderDescription() {
+        const { children, descriptionKey, t } = this.props;
```

Comment:
What children do we support here?

Reply:
For example, MuteEveryoneDialog use state dependant children.

Reply:
Hum, but I see that would include a Switch too. That's not going to fly, let's think of something else.